Practical Dependency InversionTests/MainThreadDecoratorTests.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  MainThreadTests.swift
+//  Practical Dependency InversionTests
+//
+//  Created by 游宗諭 on 2020/7/14.
+//  Copyright © 2020 Hanyu. All rights reserved.
+//
+
+import XCTest
+@testable import Practical_Dependency_Inversion
+
+class MainThreadDecoratorTests: XCTestCase {
+    
+    
+    func testCanCheckNotOnMainQueue() throws {
+        let testQueue = DispatchQueue(label: #function)
+        let await = XCTestExpectation(description: #function)
+        var result:Bool?
+        testQueue.async {
+            checkQueue(queue: .main, result: &result)
+            await.fulfill()
+        }
+        wait(for: [await], timeout: 1)
+        let returned = try XCTUnwrap(result)
+        XCTAssertFalse(returned)
+    }
+    func testCanCheckIsOnMainQueue() throws {
+        let testQueue = DispatchQueue(label: #function)
+        let await = XCTestExpectation(description: #function)
+        var result:Bool?
+        testQueue.async {
+            DispatchQueue.main.async {
+                checkQueue(queue: .main, result: &result)
+                await.fulfill()
+            }
+        }
+        wait(for: [await], timeout: 1)
+        let returned = try XCTUnwrap(result)
+        XCTAssertTrue(returned)
+    }
+    func testItemUIComposerIsOnMainQueue() throws {
+        let await = XCTestExpectation(description: #function)
+        let spec = ItemViewControllerSpecSpy()
+        let sut = MainThreadDecorator(loader: spec)
+        var result:Bool? = nil
+        sut.items {  (_) in
+            checkQueue(queue: .main, result: &result)
+            await.fulfill()
+        }
+        wait(for: [await], timeout: 1)
+        let returned = try XCTUnwrap(result)
+        XCTAssertTrue(returned)
+    }
+}
+
+
+func checkQueue(queue: DispatchQueue, result: inout Bool?){
```

Comment:
[Q]
why use inout?

Reply:
No idea why I prefer using `inout`, but I don't know why not.

Reply:
Yap, it turns out you are right! Returning the value is pretty neat!